icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/PluralRulesTest.java

```diff
@@ -929,6 +930,45 @@ public void TestKeywords() {
         }
     }
 
+
+
+    @Test
+    public void testCompactDecimalPluralKeyword() {
+        PluralRules rules = PluralRules.createRules("one: i = 0,1 @integer 0, 1 @decimal 0.0~1.5;  many: e = 0 and i % 1000000 = 0 and v = 0 or " +
+                "e != 0 .. 5;  other:  @integer 2~17, 100, 1000, 10000, 100000, 1000000, @decimal 2.0~3.5, 10.0, 100.0, 1000.0, 10000.0, 100000.0, 1000000.0, …");
+        ULocale locale = new ULocale("fr-FR");
+
+        // simple notation (no rounding from compact notation)
+        Notation simple = Notation.simple();
+        assertEquals("PluralRules select, no suppressed exponent", "one", getPluralKeyword(rules, locale, 0, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "one", getPluralKeyword(rules, locale, 1, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "other", getPluralKeyword(rules, locale, 2, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "other", getPluralKeyword(rules, locale, 100, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "many", getPluralKeyword(rules, locale, 1000000, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "other", getPluralKeyword(rules, locale, 1000001, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "other", getPluralKeyword(rules, locale, 1200000, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "other", getPluralKeyword(rules, locale, 1200001, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "many", getPluralKeyword(rules, locale, 2000000, simple));
+
+        // compact notation (rounding)
+        Notation compactLong = Notation.compactLong();
+        assertEquals("PluralRules select, suppressed exponent", "one", getPluralKeyword(rules, locale, 0, compactLong));
+        assertEquals("PluralRules select, suppressed exponent", "one", getPluralKeyword(rules, locale, 1, compactLong));
+        assertEquals("PluralRules select, suppressed exponent", "other", getPluralKeyword(rules, locale, 2, compactLong));
+        assertEquals("PluralRules select, suppressed exponent", "other", getPluralKeyword(rules, locale, 100, compactLong));
+        assertEquals("PluralRules select, suppressed exponent", "many", getPluralKeyword(rules, locale, 1000000, compactLong));
```

Comment:
This is the line that is failing according to the CI.  I suspect that it might be a problem not in your code, but in the rule string; it looks like "one" will match before "many".

Reply:
This test passes now without any extra modifications after having completed the rework on the suppressed exponent and the calculations of plural operand values.